app/RangeTest/rangetest.py

```diff
@@ -0,0 +1,230 @@
+#!/usr/bin/python
+
+#============================ adjust path =====================================
+
+import sys
+import os
+if __name__ == "__main__":
+    here = sys.path[0]
+    sys.path.insert(0, os.path.join(here, '..', '..','libs'))
+    sys.path.insert(0, os.path.join(here, '..', '..','external_libs'))
+
+#============================ imports =========================================
+
+import time
+import threading
+import traceback
+import pdb
+
+from SmartMeshSDK.IpMoteConnector      import IpMoteConnector
+from SmartMeshSDK.ApiException         import APIError, \
+                                              ConnectionError
+
+from dustCli                           import DustCli
+from SmartMeshSDK                      import sdk_version
+
+#============================ defines =========================================
+
+#============================ globals =========================================
+
+connector     = None
+range_tester  = None
+
+#============================ helpers =========================================
+
+def printExcAndQuit(err):
+
+    output  = []
+    output += ["="*30]
+    output += ["error"]
+    output += [str(err)]
+    output += ["="*30]
+    output += ["traceback"]
+    output += [traceback.format_exc()]
+    output += ["="*30]
+    output += ["Script ended because of an error. Press Enter to exit."]
+    output  = '\n'.join(output)
+
+    raw_input(output)
+    sys.exit(1)
+
+#============================ objects =========================================
+
+class RangeTester(object):
+
+    def __init__(self,connector):
+
+        # store params
+        self.connector              = connector
+        self.stationId              = 26
+        self.goOn                   = True
+        self.stats                  = [""]*16              # a list of statistic per channel
+
+
+
+    #======================== public ==========================================
+
+    def start(self):
+
+        # set the current channel to default
+        cc = 0
+
+        # loop to the infinite and beyond
+        while(self.goOn):
+            # current channel
+            cc = cc % 16
+
+            # channel mask
+            cm = 0x1<<cc
+
+            # listen for packets
+            resp = self._readPackets(cm)
+            radioRxDone = False
+
+            while not radioRxDone:
+                time.sleep(1)
+
+                # try to get last statistics
+                try:
+                    last_stats = self._getStats()
+                except APIError:
+                    continue
+
+                radioRxDone = True

```

Comment:
I don't understand the flow here. Isn't `radioRxDone` always set after 1s in your current code?